### PR TITLE
IS-36 SDK - страница настроек

### PR DESCRIPTION
**Pull request by jcbbb**

### CODE REVIEW

File: pkg/infrastructure/persistence/tab_repository.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+func (g *GormTabRepository) GetAll(ctx context.Context, params *tab.FindParams) ([]*tab.Tab, error) {
+	tx, ok := composables.UseTx(ctx)
+	if !ok {
+		return nil, service.ErrNoTx
+	}
+	q, err := helpers.ApplySort(tx, params.SortBy)
+	if err != nil {
+		return nil, err
+	}
+	if params.UserID != 0 {
```
Comment (diyor28) [inline pkg/infrastructure/persistence/tab_repository.go]: Do not overload `GetAll` method with filtering by UserId. Instead create a separate `GetUserTabs` method

File: pkg/infrastructure/persistence/tab_repository.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		q.Where("user_id = ?", params.UserID)
+	}
+	var entities []*models.Tab
+	if err := q.Find(&entities).Error; err != nil {
+		return nil, err
+	}
+
+	tabs := make([]*tab.Tab, len(entities))
```
Comment (diyor28) [inline pkg/infrastructure/persistence/tab_repository.go]: Use `mapping.MapDbModels` (however you will have to change `ToDomainTab` return signature. 
Or use `mapping.MapViewModels`

File: pkg/infrastructure/persistence/tab_repository.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		return service.ErrNoTx
+	}
+	tab := ToDBTab(data)
+	if err := tx.Save(tab).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (g *GormTabRepository) Delete(ctx context.Context, params *tab.DeleteParams) error {
```
Comment (diyor28) [inline pkg/infrastructure/persistence/tab_repository.go]: Here as well

**Review by diyor28 — CHANGES_REQUESTED** [review]

File: pkg/presentation/viewmodels/models.go [hunk 4]
```diff
@@ -116,3 +116,9 @@ type Upload struct {
 	CreatedAt string
 	UpdatedAt string
 }
+
+type Tab struct {
+	ID       string
+	Href     string
+	Position uint
```
Comment (diyor28) [inline pkg/presentation/viewmodels/models.go]: This should be a string, unless there is a good reason

**Review by diyor28 — CHANGES_REQUESTED** [review]

File: pkg/infrastructure/persistence/tab_repository.go (hunk 1 above)
Comment (jcbbb) [reply pkg/infrastructure/persistence/tab_repository.go]: What's wrong with filtering?

File: pkg/presentation/viewmodels/models.go (hunk 4 above)
Comment (jcbbb) [reply pkg/presentation/viewmodels/models.go]: Don't need this in viewmodel, removed it

File: pkg/infrastructure/persistence/tab_repository.go (hunk 1 above)
Comment (diyor28) [reply pkg/infrastructure/persistence/tab_repository.go]: I was lazy to write it out myself, so here's the same thing from ChatGPT:

image

Comment (jcbbb) [reply pkg/infrastructure/persistence/tab_repository.go]: fixed

File: pkg/infrastructure/persistence/tab_repository.go (hunk 3 above)
Comment (jcbbb) [reply pkg/infrastructure/persistence/tab_repository.go]: fixed

**Pull request merged**